src/index.ts

```diff
@@ -146,7 +146,7 @@ function fillOrders(side: string, price: number, quantity: number, userId: strin
       }
       if (asks[i].quantity > remainingQuantity) {
         asks[i].quantity -= remainingQuantity;
-        flipBalance(asks[i].userId, userId, remainingQuantity, price);
+        flipBalance(asks[i].userId, userId, remainingQuantity, asks[i].price); // replace price variable(user placed price) placed by user with ask price on order book. asks[i].
```

Comment:
Probably not the only place this change needs to be made

Reply:
updated at all positions.